Submission.java

```diff
@@ -210,6 +235,195 @@ private int getCountForStage(String item)
     }
 
     /**
+     * Count block categories for stage.
+     */
+    public void countBlockCategoriesForStage()
+    {
+        controlBlocksForStage = 0;
+        dataBlocksForStage = 0;
+        eventsBlocksForStage = 0;
+        looksBlocksForStage = 0;
+        moreBlocksBlocksForStage = 0;
+        motionBlocksForStage = 0;
+        operatorsBlocksForStage = 0;
+        penBlocksForStage = 0;
+        sensingBlocksForStage = 0;
+        soundBlocksForStage = 0;
+        JSONArray scripts = FileUtils.getJSONArrayAttribute(jsonObj, "scripts");
+        processScripts(scripts);
+    }
+
+    /**
+     * Process scripts to count blocks by category.
+     *
+     * @param array 
+     */
+    private void processScripts(JSONArray array)
+    {
+        if (array == null || array.size() == 0)
+        {
+            return;
+        }
+        
+        // If first element is a String, it is the block name.
+        // Get and count its category.
+        if (array.get(0) instanceof String)
+        {
+            String category = getCategory((String) array.get(0));
+            if (category != null)
```

Comment:
@KaraBeason The JSONArray and JSON Object stuff are a bit odd - they always return Objects, even though those Objects might actually be other things. So you have to cast them as the type they actually are to get them to behave the proper way. If you look at the example we started with (https://www.mkyong.com/java/json-simple-example-read-and-write-json/) you can see when he is reading the parsed JSON, he has to cast it as well. It won't compile otherwise :(.